src-tauri/src/farmer.rs

```diff
@@ -208,17 +210,15 @@ async fn farm(base_directory: PathBuf, farm_args: FarmingArgs) -> Result<(), any
         match result {
             Err(error) => {
                 error!("{error}");
-                error_sender
-                    .send(())
-                    .await
-                    .expect("error receiver is always listening")
+                error_sender.send(()).await.expect(
+                    "the receiver is defined in the `farming` function, it never stops listening",
```

Comment:
I understand that something somewhere might actually be holding it, but as far as this function is concerned, it is possible, very easily in fact, to make this panic. Why don't you just print an error about it, why do you need `.expect()`?

Also I don't quite understand why background task is needed here. You could return `multi_plots_farm` from this function and then call `.wait()` on the higher level where both sender and receiver sides of the channel are created and you can actually guarantee it will not panic.

Reply:
That sounds more direct than my approach, I will try that :) Thanks!

Reply:
Tried it, did not like it because of code duplicaiton. 
In the higher level one:
- we are starting `farm` instance (1),
- we start listening for errors on that `farm` instance
- if any error is sent, we are starting a new `farm` instance (2)

so, this `wait()` mechanism and the long code for sending error messages after that are duplicated. 
I can put these duplicated code pieces into a function, but then it is no different than before. 

And this code duplication is around 20 lines, making the higher level function unreadable.

----

The reason I wanted to utilize `expect` instead of printing an error about it is:
I think it is not possible for the `receiver` to stop listening, and if that occurs, something should have been really off in the higher level function. In this case, a new `farm` instance won't be created successfully, and the main functionality of the desktop app will be hindered. That's why, panicking seemed like a good choice.

Reply:
Something like this?:
```rust
loop {
    if let Err(error) = start_farm.await {
        // TODO
    }
}
```

Shouldn't have much code duplication, just a matter or restructuring it.

> I think it is not possible for the receiver to stop listening, and if that occurs, something should have been really off in the higher level function

My issue with this is that it is not possible to guarantee just within this function. It is ideal for functions to guarantee lack of panics statically.